### PR TITLE
[Dtype Function] fix aten.div.Tensor_mode's dtype function

### PR DESCRIPTION
**Pull request by qingyunqu**

### CODE REVIEW

File: projects/pt1/python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py [hunk 1]
```diff
@@ -2391,31 +2403,24 @@ def aten〇div〇Tensor〡dtype(self_rank_dtype: Tuple[int, int], other_rank_dty
...
     promoted_dtype = promote_dtypes(ranks, dtypes)
     if is_complex_dtype(promoted_dtype) or \
-       (is_float_dtype(promoted_dtype) and promoted_dtype != torch.float32):
+       (is_float_dtype(promoted_dtype) and promoted_dtype != torch.float32) or \
+        (rounding_mode is not None and rounding_mode == "trunc"):
         return promoted_dtype
     else:
         return torch.float32
 
-@check_dtype_function(_check_two_tensor_op(input_error_types={torch.complex64, torch.complex128}, output_error_types={torch.bool}))
```
Comment (ramiro050) [inline projects/pt1/python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py:2407]: What's the reason for moving this one?

File: projects/pt1/python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py [hunk 2]
```diff
@@ -2391,31 +2403,24 @@ def aten〇div〇Tensor〡dtype(self_rank_dtype: Tuple[int, int], other_rank_dty
     else:
         return torch.float32
 
-@check_dtype_function(_check_two_tensor_op(rounding_mode=None))
+@check_dtype_function(_check_two_tensor_op(rounding_mode=None) +
+                      _check_two_tensor_op(input_error_types={torch.complex64, torch.complex128}, output_error_types={torch.bool}, rounding_mode="floor") +
+                      _check_two_tensor_op(rounding_mode="trunc"))
 def aten〇div〇Tensor_mode〡dtype(self_rank_dtype: Tuple[int, int], other_rank_dtype: Tuple[int, int], rounding_mode: Optional[str]) -> int:
+    if rounding_mode is not None and rounding_mode == "floor":
+        return aten〇floor_divide〡dtype(self_rank_dtype, other_rank_dtype)
```
Comment (qingyunqu) [inline projects/pt1/python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py:2411]: here

File: projects/pt1/python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py (hunk 1 above)
Comment (ramiro050) [reply projects/pt1/python/torch_mlir/dialects/torch/importer/jit_ir/build_tools/abstract_interp_lib_gen.py:2407]: Ah, I missed that.

**Review by ramiro050 — APPROVED** [review]
LGTM!

**Pull request merged**